src/sema/statements.rs

```diff
@@ -309,12 +333,20 @@ fn statement(
                     Some(&var_ty),
                 )?;
 
+                used_variable(ns, &expr, symtable);
+
                 Some(cast(&expr.loc(), expr, &var_ty, true, ns, diagnostics)?)
             } else {
                 None
             };
 
-            if let Some(pos) = symtable.add(&decl.name, var_ty.clone(), ns) {
+            if let Some(pos) = symtable.add(
+                &decl.name,
+                var_ty.clone(),
+                ns,
+                initialized,
```

Comment:
You could replace this with `initializer.is_some()`

Reply:
Done